website/server/controllers/api-v3/hall.js

```diff
@@ -177,8 +177,8 @@ api.updateHero = {
       _.set(hero, updateData.itemPath, updateData.itemVal); // Sanitization at 5c30944 (deemed unnecessary)
     }
 
-    if (updateData.auth && _.isBoolean(updateData.auth.blocked)) {
-      hero.auth.blocked = updateData.auth.blocked;
+    if (updateData.auth && _.isBoolean(updateData.auth.blocked) && updateData.auth.blocked === true) {
```

Comment:
If the hero was already blocked, do you still want to change their sleep state?
This just checks what update was sent, but not what the current state of the hero is. 

(Maybe not, just a thought!)